scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -196,115 +197,128 @@ public int schedule() {
 
             // ask the policy all the tasks to be schedule according to the jobs list.
 
-            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = currentPolicy.getOrderedTasks(descriptors);
+            LinkedList<EligibleTaskDescriptor> fullListOfTaskRetrievedFromPolicy = currentPolicy.getOrderedTasks(descriptors);
 
             //if there is no task to scheduled, return
-            if (taskRetrievedFromPolicy == null || taskRetrievedFromPolicy.isEmpty()) {
+            if (fullListOfTaskRetrievedFromPolicy == null || fullListOfTaskRetrievedFromPolicy.isEmpty()) {
                 return numberOfTaskStarted;
             }
 
-            if (logger.isDebugEnabled()) {
-                logger.debug("eligible tasks : " + taskRetrievedFromPolicy);
-            }
-
             schedulingService.unlockJobsToSchedule(toUnlock.values());
             toUnlock = null;
 
-            updateVariablesForTasksToSchedule(jobMap, taskRetrievedFromPolicy);
+            int subTasksListFromInclusive = 0;
```

Comment:
if these are range integer values, they should not be named subTaskList...

You could make use of the guava class Range to avoid using two variables

I would separate the logic to extract a sublist from the full list in a dedicated class. This way you can unit test it, which seems important.

Reply:
I never used guava but can have a look when I will have time